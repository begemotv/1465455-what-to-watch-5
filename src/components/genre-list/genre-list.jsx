import React from "react";
import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';
import {connect} from "react-redux";

import {filmPropTypes} from "../../prop-types";
import {ActionCreator} from "../../store/action";

const GenreList = (props) => {
  const {
    activeGenre,
    filmsAll,
    handleGenreTabClick,
    id
  } = props;

  let genreList = filmsAll
    .map((item) => item.genre)
    .filter((item, index, arr) => arr.indexOf(item) === index)
    .slice(0, 9);

  genreList.unshift(`All genres`);

  return (
    <ul className="catalog__genres-list">
      {genreList.map((genre) => (
        <li
          key={`genre-${genre}`}
          className={`catalog__genres-item ${genre === activeGenre ? `catalog__genres-item--active` : ``}`}>
          <Link
            to="#"
            className="catalog__genres-link"
            onClick={
              (evt) => {
                evt.preventDefault();
                handleGenreTabClick(genre, id);
              }}
          >{genre}</Link>
        </li>
      ))}
    </ul>
  );
};

GenreList.propTypes = {
  activeGenre: PropTypes.string.isRequired,
  filmsAll: PropTypes.arrayOf(
      PropTypes.shape(filmPropTypes)
  ).isRequired,
  handleGenreTabClick: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired
};

const mapStateToProps = (state) => ({
  activeGenre: state.activeGenre,
  filmsAll: state.filmsAll
});

const mapDispatchToProps = (dispatch) => ({
  handleGenreTabClick(genre, id) {
    dispatch(ActionCreator.changeGenre(genre));
    dispatch(ActionCreator.filterByGenre(genre, id));
  },
});

export {GenreList};
export default connect(mapStateToProps, mapDispatchToProps)(GenreList);
