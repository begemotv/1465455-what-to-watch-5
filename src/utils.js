export const nameRating = (rating) => {
  let ratingName = ``;
  switch (true) {
    case rating >= 0 && rating < 3:
      ratingName = `Bad`;
      break;
    case rating >= 3 && rating < 5:
      ratingName = `Normal`;
      break;
    case rating >= 5 && rating < 8:
      ratingName = `Good`;
      break;
    case rating >= 8 && rating < 10:
      ratingName = `Very good`;
      break;
    case rating === 10:
      ratingName = `Awesome`;
      break;
    default:
      ratingName = `Not rated`;
  }
  return ratingName;
};

