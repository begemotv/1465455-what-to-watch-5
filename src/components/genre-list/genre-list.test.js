import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";

import {GenreList} from "./genre-list";
import {noop} from "../../utils";

const activeGenreMock = `Action`;
const genreListMock = [`Action`, `Drama`, `Comedy`, `Thriller`, `Horror`];

test(`GenreList renders correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter >
          <GenreList
            activeGenre={activeGenreMock}
            genreList={genreListMock}
            handleGenreTabClick={noop}
          />,
        </MemoryRouter >
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
