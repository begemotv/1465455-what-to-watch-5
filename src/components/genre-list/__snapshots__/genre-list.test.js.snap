// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList renders correctly 1`] = `
Array [
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="/"
        onClick={[Function]}
      >
        Action
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="/"
        onClick={[Function]}
      >
        Drama
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="/"
        onClick={[Function]}
      >
        Comedy
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="/"
        onClick={[Function]}
      >
        Thriller
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="/"
        onClick={[Function]}
      >
        Horror
      </a>
    </li>
  </ul>,
  ",",
]
`;
